task8/src/main/webapp/resources/scripts/postEvents.js

```diff
@@ -0,0 +1,168 @@
+        if (deletePost) {
+            deletePost.addEventListener('click',
+                evt => {
+                    modals.createDeleteModal(postId);
+                    evt.stopPropagation();
+                });
+        }
+
+        const editPost = postElement.getElementsByClassName('edit').item(0);
+        if (editPost) {
+            editPost.addEventListener('click',
+                evt => {
+                    modals.createEditModal(postId);
+                    evt.stopPropagation();
+                })
+        }
+    }
+
+    async setNewPostEventListener(postAreaElement) {
+        const add = postAreaElement.getElementsByTagName("form").item(0);
+        if (add) {
+            add.addEventListener('submit',
+                evt => {
+                    try {
+                        postEvent._postNewPost();
+                    } catch (error) {
+                        modals.createErrorModal("Server error");
+                    }
+                    evt.stopPropagation();
+                }
+            );
+        }
+        const addPhoto = postAreaElement.getElementsByTagName("input").item(0);
+        if (addPhoto) {
+            addPhoto.addEventListener('click',
+                evt => {
+                    postEvent.drawPreviewPhoto('img-file', 'file-preview');
+                    evt.stopPropagation();
+                }
+            );
+        }
+    }
+
+    async deletePost(url, id) {
+        return await fetch(url, {
+            method: 'DELETE',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({id: id})
+        }).then(response => response.status);
+    }
+
+    async editPost(url, id, editFields) {
+        return await fetch(url, {
+            method: "POST",
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                id: id,
+                description: editFields.description,
+                discount: editFields.discount,
+                hashTags: editFields.hashTags,
+                validateUntil: editFields.validateUntil,
+            })
+        }).then(response => response.status);
+    }
+
+    async postLike(postId, url) {
+        return await fetch(url, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({id: postId, author: view.getUser()})
+        }).then(response => response.status);
+    }
+
+    async _updateLike(postId) {
+        let response = await postEvent.postLike(postId, "/post/update/like");
+        if (response !== 200) {
+            modals.createErrorModal("Error updating like");
+            return;
+        }
+        await view.postViewer.pressLike(postId);
+    }
+
+    async _postNewPost() {
+        let newPost =await addPostEvent.fillNewPostData();
```

Comment:
Here `await` is used in a right way, but again, model and controller code is mixed